### DISCUSSION TITLE
TPI3 - Eventos Repetitivos Infinitos

### DISCUSSION THREAD

**Discussion by guidoenr** [dds-jv/foro#286] [Consultas de objetos] [answered]
Buenas que tal, viendo un poco el foro vi que hay una pregunta muy parecida (https://github.com/dds-jv/foro/discussions/278) a la que voy a realizar a continuacion , donde la idea es la misma.

Mi duda como comenta el compañero  es el **como modelar los eventos que tienen repeticiones** para poder jugar con las comparaciones, es decir:

supongamos un **Evento Repetitivo** que inicia un **`01 - 01 - 2021`** y se repite cada **4 dias**  , osea, estaria tambien en las fechas : 
```
01 - 05 - 2021 
01 - 09 - 2021
01 - 13 - 2021
NN - NN - NN
...INF... + 4
```
y supongamos otro **Evento Simple** que inicia **`09 - 01 - 2021`** (obviamente sin repeticiones) 

si quisiera ver los eventos entre las fechas **`08 - 01 - 2021`** y **`09 - 01 - 2021`** realmente lo que pasaria es que deberia devolverme los dos eventos que nombre arriba, puesto que el mismo se repite.. todo bien, hago un while/for o lo que fuese y lo obtengo.. pero claro, aca la complejidad algoritmica es bajisima porque el evento esta re cerca, pero suponete que el evento se esta haciendo desde **1965** y justo tambien pasa en la fecha que tengo el evento simple.. ahi que hago? porque ni hablar lo que tardaria en pasar todos los años hasta pegarle a un **`true`** unicamente **sumando 4 DIAS** 
y bueno, ni hablar cuando comparemos dos eventos repetitivos, o si estan solapados, o el peor de los casos.. que nunca se cruzen y que  el while/for no sepa cuando parar..

el algoritmo lo veo muy lento, con una complejidad algoritmica muy alta la verdad.. asi que se me hace que lo estoy pensando mal, o tal vez esta no es la forma de encararlo.

Me vendria bien un centro. Gracias de antemano

PD: capaz existe alguna libreria magica que pueda hacer una prediccion con un algoritmo menos costoso

**Comment by gastonprieto** [comment, marked as answer]
me confundio un toque el formato, mes - dia - año XD, si no usan el que usamos acá, aclaren :P

no hace falta resolver la parte algoritmica para bajar la complejidad. Primero hacer que ande y este bien diseñado, tema performance se ataca mas adelante, pero me imagino que dado que las repeticiones son en dias vos podrias hacer una cuenta que sume un multiplo dia dias anterior a la distancia al evento mas cercano al rango. 

No lo pense demasiado, porque el foco acá es como modelan las abstracciones y delegan las responsabilidades, no el algoritmo, no importa que no sea eficiente.
